mobile-sdk/plain-webview.md

```diff
@@ -0,0 +1,559 @@
+
+Sometimes, a payment flow calls for launching an external application, like BankID or Swish. A web page does this by opening a url that is handled by the app in question. To accommodate for this, we extend the "Listening for Navigations" approach above. If you opted for JavaScript hooks for completion, you will now need to add a navigation listener for external apps.
+
+Determining whether a url should launch an external app is straightforward, though on Android it involves a bit of a judgement call. Let us take a look at the arguably simpler iOS case first.
+
+### iOS
+
+You cannot query the system for an arbitrary url to see if it can be opened – this is a deliberate privacy measure. What can be done, and what also happens to be exactly what we want to do, is to attempt to open a url and receive a callback telling us whether it succeeded. Nowadays, the recommended way of opening external applications is to use Universal Links, anyway, which are, on the surface, indistiguishable from web links.
+
+{:.code-header}
+**iOS**
+```swift
+    func webView(
+        _ webView: WKWebView,
+        decidePolicyFor navigationAction: WKNavigationAction,
+        decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
+    ) {
+        // Check for completeUrl and cancelUrl here, if applicable.
+
+        if let url = navigationAction.request.url {
+            openInExternalApp(
+                url: url,
+                decisionHandler: decisionHandler
+            )
+        } else {
+            // N.B. This should never happen.
+            decisionHandler(.allow)
+        }
+    }
+
+    private func openInExternalApp(
+        url: URL,
+        decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
+    ) {
+        // First, check for universal link
+        UIApplication.shared.open(
+            url,
+            options: [.universalLinksOnly: true]
+        ) { universalLinkOpened in
+            if universalLinkOpened {
+                // Url was opened in external app, so do not open it in WKWebView.
+                decisionHandler(.cancel)
+            } else {
+                // Url was not opened in external app, see if WKWebView can handle it.
+                if let scheme = url.scheme,
+                    WKWebView.handlesURLScheme(scheme) {
+                    // Regular http(s) url, proceed.
+                    decisionHandler(.allow)
+                } else {
+                    // Custom-scheme url. Try to open it.
+                    UIApplication.shared.open(url)
+                    // Cancel the navigation regardless of the result,
+                    // as WKWebView does not know what to do with the url anyway.
+                    decisionHandler(.cancel)
+                }
+            }
+        }
+    }
+```
+
+### Android
```

Comment:
Codacy found an issue: [[no-duplicate-headings] Do not use headings with similar content (233:1)](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5758572)